Symfony/CS/Console/Command/SelfUpdateCommand.php

```diff
@@ -56,28 +64,46 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 1;
         }
 
-        $remoteTag = $this->getRemoteTag();
-
+        $remoteTag = $this->getLatestTag();
         if (null === $remoteTag) {
             $output->writeln('<error>Unable to determine newest version.</error>');
 
-            return;
+            return 0;
         }
 
-        if ('v'.$this->getApplication()->getVersion() === $remoteTag) {
+        $currentVersion = 'v'.$this->getApplication()->getVersion();
+        if ($currentVersion === $remoteTag) {
             $output->writeln('<info>php-cs-fixer is already up to date.</info>');
 
-            return;
+            return 0;
         }
 
-        $remoteFilename = $this->buildVersionFileUrl($remoteTag);
+        $remoteVersionParsed = $this->parseVersion($remoteTag);
+        $currentVersionParsed = $this->parseVersion($currentVersion);
+
+        if ($remoteVersionParsed[0] > $currentVersionParsed[0] && true !== $input->getOption('force')) {
+            $output->writeln(sprintf('<info>A new major version of php-cs-fixer is available</info> (<comment>%s</comment>)', $remoteTag));
+            $output->writeln(sprintf('<info>Before upgrading please read</info> https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/%d.%d/UPGRADE.md', $remoteTag[0], $remoteTag[1]));
```

Comment:
`$remoteTag` is a string, eg `v1.12.3`. accessing it's idxs is not good idea.
Didn't you want to use `$remoteVersionParsed` here instead ?